srtcore/core.cpp

```diff
@@ -10824,15 +10828,21 @@ void CUDT::checkRexmitTimer(const steady_clock::time_point& currtime)
     // Otherwise it might still be sent regulary.
     bool retransmit = false;
     const int32_t unsent_seqno = CSeqNo::incseq(m_iSndCurrSeqNo);
+
+    // TODO: [MAX] Replace the condition with
+    // if (getFlightSpan() > 0 && (!is_laterexmit || m_pSndLossList->getLossLength() == 0))
+    // Do we need two retransmission modes? The only thing that matters is if the peers sends NAK periodically.
+
     // IF:
-    // - LATEREXMIT
+    // - LATEREXMIT (File Mode)
     // - flight window == 0
     // - the sender loss list is empty (the receiver didn't send any LOSSREPORT, or LOSSREPORT was lost on track)
+    // OR IF:
+    // - FASTREXMIT (Live Mode)
+    // - flight window > 0
     if ((is_laterexmit && unsent_seqno != m_iSndLastAck && m_pSndLossList->getLossLength() == 0)
-    // OR:
-            // - FASTREXMIT
-            // - flight window > 0
-         || (is_fastrexmit && getFlightSpan() != 0))
+        // TODO: [MAX] Previously was CSeqNo::seqoff(m_iSndLastAck, unsent_seqno) > 0
+         || (is_fastrexmit && getFlightSpan() > 0))
```

Comment:
Is `getFlightSpan()` now not guaranteed to never return a value less than 0?

Reply:
It is guaranteed. Just `> 0` is cleaner in the code. The `getFlightSpan()` returns a signed integer, so if you don't know the implementation of `getFlightSpan()`, then `!= 0` is confusing.